### PR TITLE
feat(filter-field): Allow freeText suggestions to be async.

### PR DESCRIPTION
**Pull request by danielkaneider**

### CODE REVIEW

**Review by ffriedl89 — APPROVED** [review]
This is awesome! Thx @danielkaneider for this! 🚀

File: libs/barista-components/filter-field/src/filter-field-default-data-source.ts [hunk 1]
```diff
@@ -20,15 +20,15 @@ import { isDefined, isObject } from '@dynatrace/barista-components/core';
...
-  dtGroupDef,
   dtFreeTextDef,
-  dtRangeDef,
+  dtGroupDef,
   dtMultiSelectDef,
+  DtNodeDef,
+  dtOptionDef,
+  dtRangeDef,
+  isDtAutocompleteDef,
+  isDtGroupDef,
```
Comment (tomheller) [inline libs/barista-components/filter-field/src/filter-field-default-data-source.ts:31]: Nitpick.
If it is not too much trouble, could you change these back. There is no change here other than the automatic import sorter of your IDE.

File: libs/barista-components/filter-field/src/types.ts [hunk 2]
```diff
@@ -342,6 +344,31 @@ export function isDtFreeTextDef<D = unknown, S = unknown>(
   return isDtNodeDef<D>(def) && !!(def.nodeFlags & DtNodeFlags.TypeFreeText);
 }
 
+export function isAsyncDtFreeTextDef<D>(
```
Comment (tomheller) [inline libs/barista-components/filter-field/src/types.ts:347]: Please add JSDoc comment.

File: libs/barista-components/filter-field/src/types.ts [hunk 3]
```diff
@@ -342,6 +344,31 @@ export function isDtFreeTextDef<D = unknown, S = unknown>(
...
+  option: DtOptionDef;
+} {
+  return (
+    isDtFreeTextDef<D>(def) &&
+    isDtOptionDef<D>(def) &&
+    Boolean(def.freeText?.async)
+  );
+}
+
+export function isAsyncDtOptionDef<D>(
```
Comment (tomheller) [inline libs/barista-components/filter-field/src/types.ts:360]: Please add JSDoc comment.

File: libs/barista-components/quick-filter/src/quick-filter-default-data-source.ts [hunk 4]
```diff
@@ -16,21 +16,21 @@
...
+  dtRangeDef,
   isDtAutocompleteDef,
-  DtFilterFieldDefaultDataSourceAutocomplete,
-  DtFilterFieldDefaultDataSourceSimpleGroup,
-  DtFilterFieldDefaultDataSourceGroup,
-  DtFilterFieldDefaultDataSourceFreeText,
-  DtFilterFieldDefaultDataSourceRange,
-  dtAutocompleteDef,
   isDtGroupDef,
-  DtFilterFieldDefaultDataSourceMultiSelect,
```
Comment (tomheller) [inline libs/barista-components/quick-filter/src/quick-filter-default-data-source.ts:33]: Nitpick, please revert the import sort.

**Review by tomheller — APPROVED** [review]

**Review by tomheller — APPROVED** [review]
Perfect, thank you

**Review by thomaspink — APPROVED** [review]

**Comment by tomheller** [comment]
Cherry picked into 9.x, new head 73bdeea2ee5529381b69652b0c4eddb73be24ae2

**Pull request merged**